src/app/core/services/logging.service.ts

```diff
@@ -16,19 +17,30 @@ export class LoggingService {
     }
   }
 
+  saveToLocalStorage(...params: any[]) {
+    const oldestSessionLogRegex = /(AppConfig,.*\n[\S\s]*)(?=AppConfig,.*)/g;
+    const existingLog = localStorage.getItem(this.LOG_KEY);
+    const newLog = `${existingLog ? `${existingLog}\n` : ''}${params.toString()}`.replace(oldestSessionLogRegex, '');
+    localStorage.setItem(this.LOG_KEY, newLog);
+  }
```

Comment:
Based on our current logs in console, `AppConfig, ...` is the first item logged when the app is started / refreshed. So at the moment it wipes out all the content between 2 `AppConfigs, ...` so that the latest session information is stored.

Reply:
If the app crashes and a student starts it up again to download the logs, will this mean that the previous session data (and what was caused the crash) would be removed?

Reply:
At the moment yes, I'm in the works of implementing storing multiple sessions and a download button in a separate PR.

Reply:
Included the check multiple sessions 👍